### PR TITLE
initial commit of vault pack

### PR DESCRIPTION
**Pull request by st33v**
This is pretty rough but the pack can get and set values in the vault and can then read them back. I've also started work on getting and setting policies.

### CODE REVIEW

**Comment by dzimine** [comment]
Steve, great, thank you for the contribution! 

Please add README.md with at least a sentence to clarify that it's HarshiCorp Vault (may be hvac would be a more explicit pack name), and we'll merge.

Consider using recently-added unit-testing framework to simplify your development https://docs.stackstorm.com/development/pack_testing.html - there are few examples in st2contrib.

Regards, Dmitri.

File: packs/vault/actions/write.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    path:
+        type: string
+        description: "Path to the Vault secrets"
+        required: true
+        position: 0
+    values:
+        type: string
+        description: "Keys and values to write in Vault ({'key':'value','key2': 'value2'}"
+        required: true
+        position: 1
```
Comment (manasdk) [inline packs/vault/actions/write.yaml]: new line :)

File: packs/vault/actions/delete.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name: delete
```
Comment (manasdk) [inline packs/vault/actions/delete.yaml]: suggest to start with `---`

File: packs/vault/actions/delete.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from lib import action
+
+class VaultDeleteAction(action.VaultBaseAction):
```
Comment (manasdk) [inline packs/vault/actions/delete.py]: 2 lines between import and class definition.

**Comment by manasdk** [comment]
Great start!

Mostly cosmetic suggestions. We could merge this as soon as those are picked up and keep iterating.

**Comment by emedvedev** [comment]
@st33v: hi! Are you going to keep working on the pack? We'd really like to see it completed and merged, so let me know if we can help!

**Comment by st33v** [comment]
sorry... i was working on this for work and then I was reprioritized. I might resume work on it in the future but have no immediate need so I'll probably not be able to allocate time.

**Comment by emedvedev** [comment]
It's fine, don't worry! Mind if I pick it up and bring to completion with some cosmetic/syntax changes some time later?

**Comment by st33v** [comment]
thanks @emedvedev !

**Comment by emedvedev** [comment]
I'm closing this PR because your pack is live and operational in the main contrib repo now :) Thanks for the work!

**Pull request closed without merging**